sdk/storage/faq.md

```diff
@@ -0,0 +1,374 @@
+# Frequently-asked Questions, Common Mistakes and Best Practices
+
+**API Usage**
+
+[How to use list-operations? Why doesn't list-operations return all blobs in a container?](#how-to-use-list-operations-why-doesnt-list-operation-return-all-blobs-in-a-container)
+
+[Can I specify an API-version other than the default one?](#can-i-specify-an-api-version-other-than-the-default-one)
+
+[How can I set a custom UserAgent string?](#how-can-i-set-a-custom-useragent-string)
+
+[What encoding is used for input and output?](#what-encoding-is-used-for-input-and-output)
+
+[How can I check if a blob exists? Is there a function `Exists()` like in legacy SDK?](#how-can-i-check-if-a-blob-exists-is-there-a-function-exists-like-in-legacy-sdk)
+
+[What is the difference between `BlockBlobClient::Upload()` and `BlockBlobClient::UploadFrom()`? When should I use one vs the other?](#what-is-the-difference-between-blockblobclientupload-and-blockblobclientuploadfrom-when-should-i-use-one-vs-the-other)
+
+[How to efficiently upload large amount of small blobs?](#how-to-efficiently-upload-large-amount-of-small-blobs)
+
+[How to ensure data integrity with transactional checksum?](#how-to-ensure-data-integrity-with-transactional-checksum)
+
+
+**Design and Concepts**
+
+[Thread safety guarantees](#thread-safety-guarantees)
+
+[Http pipeline and policies](#http-pipeline-and-policies)
+
+[Will the SDK retry failed requests? What is the default retry logic? How can I customize retry behavior?](#will-the-sdk-retry-failed-requests-what-is-the-default-retry-logic-how-can-i-customize-retry-behavior)
+
+
+**Troubleshooting**
+
+[Why do I see 503 Server Busy errors? What should I do in this case?](#why-do-i-see-503-server-busy-errors-what-should-i-do-in-this-case)
+
+[How to troubleshoot 403 errors?](#how-to-troubleshoot-403-errors)
+
+## How to use list-operations? Why doesn't list-operation return all blobs in a container?
+
+C++ SDK applies a different design for pageable operations from other languages.
```

Comment:
I am not sure how useful such a statement is. It makes it seem like all other languages do it one way and only C++ does it differently. Is that true? We also don't mention anything about why it's different and why the fact that it is different is even relevant. I think it would be better to wordsmith this a bit.

```suggestion
The C++ SDK applies a different design for pageable operations from other languages.
```